### ISSUE TITLE
Modernize and improve bundle size

### ISSUE THREAD

**Issue by kantorcodes** [hiero-ledger/hiero-sdk-js#1719]
### Problem

At present, the bundle is a whooping 1.4 megabytes big, or 296KB. 

For most modern webapps that have to stay competitive and fast, this is a huge impediment. Users are forced to download a lot of unneeded JavaScript. 

![5C60F757-66C2-4D8E-A91C-17933AC378F7](https://github.com/hashgraph/hedera-sdk-js/assets/6068672/b45ccc40-7e3c-4456-beb4-4846ac69dd66)

### Solution

- Modernize, and break out the bundle into multiple parts using tools like rollup. 
- Potentially migrate to TypeScript instead of Vanilla JS for improved type handling.

Alternatively
- Create a "lean" version of the SDK that can cover most use cases for a frontend dApp, and a separate SDK for backend NodeJS applications.

**Comment by ochikov** [comment]
Hey @kantorcodes,
Thank you for your feedback.
For sure we need to work in that direction. This means a lot of breaking changes. Keep in mind that such kind of changes might be implemented and included in v3 of the SDK. I will keep you updated as well about our plans and what might be done for your suggestions.

**Comment by kantorcodes** [comment]
> Hey @kantorcodes, Thank you for your feedback. For sure we need to work in that direction. This means a lot of breaking changes. Keep in mind that such kind of changes might be implemented and included in v3 of the SDK. I will keep you updated as well about our plans and what might be done for your suggestions.

Thanks! Keen to hear how it goes

**Comment by justynspooner** [comment]
Are there any updates on this? The @hashgraph/hedera-wallet-connect package would benefit greatly from being able to tree shake and optimise bundles for client side browser usage.

**Comment by rbarkerSL** [comment]
@SimiHunjan is this still a relevant ticket?

**Comment by SimiHunjan** [comment]
yes this ticket is still relevant just haven't had an opportunity based on other priorities. @ivaylonikolov7 have you investigated this at all recently?

**Comment by ivaylonikolov7** [comment]
Would love to take a look at that. Might be a good topic for tomorrow's sprint.

**Comment by ivaylonikolov7** [comment]
We’ve made significant progress on reducing the SDK bundle size. After testing various tools, including Vite, Webpack, and Rollup, we’ve implemented the following changes:

 - Integrated [Rollup](https://rollupjs.org/) 
 - Enabled tree-shaking
 - Removed large dependencies such as node-forge, which was increasing the transaction size by ~500 KB
 - Eliminated duplicate dependencies that were taking up excessive space in the SDK

As a result, we’ve successfully reduced the SDK bundle size. For instance, importing an AccountCreateTransaction now results in a download that’s approximately 800 KB smaller. This is thanks to code minification and the removal of heavy packages, which were replaced with either native code or lighter alternatives.

Additionally, if only the Mnemonic functionality is needed, it will now only download the relevant parts, reducing the bundle size from ~2 MB to around 500 kbs.

We are working on an additional optimization for the proto package, which is expected to further reduce the bundle size by approximately 300-400 KB. This is still a work in progress and is in the initial phase, requiring testing, implementation, and review.

This change will be included in the next JS SDK beta release. We’d appreciate your input on this.
